### PR TITLE
Adding Mapbox location and offline mode and various improvements

### PR DESCRIPTION
**Pull request by macintosh1398**
The PR improves Mapbox Search View in different manner. First, I improved the layout to have a view that is centered on any phone. I made some changes to have an offline compatible search view (not all functionalities can be used offline). I also added the location permission management, to be able to select the closest matching location when the user select a category (hotel, café, ...). Various resources were refactored.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -61,7 +61,8 @@ dependencies {
     implementation 'com.google.firebase:firebase-firestore-ktx'
     implementation 'com.google.firebase:firebase-auth:21.0.4'
     implementation 'com.firebaseui:firebase-ui-auth:8.0.1'
-    implementation 'com.mapbox.search:mapbox-search-android-ui:1.0.0-beta.29'
+    implementation 'com.mapbox.search:mapbox-search-android-ui:1.0.0-beta.30'
+    implementation 'com.google.android.gms:play-services-location:19.0.1'
```
Comment (Matathiasq13824) [inline app/build.gradle:65]: As you already change build.gradle, could you remove the warnings ?

File: gradle.properties [hunk 2]
```diff
@@ -22,7 +22,6 @@ kotlin.code.style=official
 # resources declared in the library itself and none from the library's dependencies,
 # thereby reducing the size of the R class for that library
 android.nonTransitiveRClass=true
-MAPBOX_DOWNLOADS_TOKEN=sk.eyJ1IjoibXVsdGltYXR1bSIsImEiOiJjbDMwNWZsODgxN2dyM2Nqenk2dGt0NzFtIn0.Wgj-43xPKrBmAP7TkzFGBQ
```
Comment (Matathiasq13824) [inline gradle.properties:25]: Why did you remove this token ?

File: settings.gradle [hunk 3]
```diff
@@ -9,11 +9,7 @@ pluginManagement {
                 basic(BasicAuthentication)
             }
             credentials {
-                // Do not change the username below.
-                // This should always be `mapbox` (not your username).
```
Comment (Matathiasq13824) [inline settings.gradle:13]: Why did you delete these comments ?

**Review by Matathiasq13824 — COMMENTED** [review]
Good job, but I have some questions about these changes

File: app/build.gradle (hunk 1 above)
Comment (macintosh1398) [reply app/build.gradle:65]: Good point, done

File: gradle.properties (hunk 2 above)
Comment (macintosh1398) [reply gradle.properties:25]: As it is used in the project this line is useless.

**Review by Matathiasq13824 — APPROVED** [review]

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/SearchLocationActivity.kt [hunk 4]
```diff
@@ -123,4 +148,15 @@ class SearchLocationActivity : AppCompatActivity() {
...
+
+    private companion object {
+        private const val PERMISSIONS_REQUEST_LOCATION = 0
+
+        fun Context.isPermissionGranted(permission: String): Boolean {
+            return ContextCompat.checkSelfPermission(
+                this,
+                permission
+            ) == PackageManager.PERMISSION_GRANTED
+        }
```
Comment (ValentinAebi) [inline app/src/main/java/com/github/multimatum_team/multimatum/activity/SearchLocationActivity.kt:160]: Is there a particular reason why you used an extension method here?

**Review by ValentinAebi — APPROVED** [review]
